numba/parfors/parfor.py

```diff
@@ -4145,13 +4217,33 @@ def get_user_varname(v):
     for uses in p2_usedefs.usemap.values():
         p2_uses |= uses
 
-    if not p1_body_defs.isdisjoint(p2_uses):
-        dprint("try_fuse: parfor2 depends on parfor1 body")
-        msg = ("- fusion failed: parallel loop %s has a dependency on the "
-                "body of parallel loop %s. ")
-        report = FusionReport(parfor1.id, parfor2.id,
-                                msg % (parfor1.id, parfor2.id))
-        return None, report
+    overlap = p1_body_defs.intersection(p2_uses)
+    # overlap are those variable defined in first parfor and used in the second
+    if len(overlap) != 0:
+        is_safe = True # Assume safe until proven otherwise.
```

Comment:
Can it default to be unsafe instead for a more conservative behavior?

It seems it just need all `overlap` variables to be of `ArrayCompatible` and in `p2arraynotindexed`

Reply:
I can do the following:

unsafe_var = [not isinstance(typemap[x], types.ArrayCompatible) or x in p2arraynotindexed for x in overlap]
if any(unsafe_var):

That is simpler looking but this does have potential to be slower than what I wrote that gives up as soon as it finds the first problem.  The new line above would potentially compute some useless stuff but the amount of work is trivial so I've checked in this change.  See if you like it.

Reply:
Would a generator comprehension solve this problem? If you change the brackets to parentheses, then the body will be lazily evaluated by `any()`: `unsafe_var = (not isinstance(typemap[x], types.ArrayCompatible) or x in p2arraynotindexed for x in overlap)`

Reply:
Thanks.  Done.